Update build to gradle

robbavey opened the pull request:
Update to use gradle to avoid jar-dependencies based weirdness with old version of jruby, and for consistency with other plugins using java dependencies.

sihil commented on build.gradle:
```diff
@@ -0,0 +1,97 @@
+  compile 'com.amazonaws:aws-java-sdk-dynamodb:1.11.14'
+  compile 'com.amazonaws:aws-java-sdk-s3:1.11.14'
+  compile 'com.amazonaws:aws-java-sdk-kms:1.11.14'
+  compile 'com.amazonaws:aws-java-sdk-core:1.11.16'
```
This line is duplicated at line 48.

sihil replied:
Also there is a mix of `1.11.14` and `1.11.16` - is there a reason for this?

robbavey commented:
@sihil Thanks!. 

I've fixed up the `build.gradle` to remove the dupes, and added the suggested files to the `.gitignore' file. 

The dependencies in the gradle file are essentially taken from the previous build with no attempt, in this PR, to update any of them, bar the `commons-logging` dep, which was blocking the plugin from building. The intent is to keep this PR as passive as possible, and introduce dependency changes in subsequent PRs.

webmat commented on build.gradle:
```diff
@@ -0,0 +1,94 @@
+  compile 'org.apache.httpcomponents:httpcore:4.4.4'
+  compile 'com.fasterxml.jackson.core:jackson-annotations:2.6.0'
+  compile 'com.google.protobuf:protobuf-java:2.6.1'
+}
```
I still know very little about Gradle, so I'm not the best person to review this PR.

But I see the two jar dependencies deleted from the gemspec in here. That's good.

Now my comment: were we not pinning everything else at all? If so, this is a great improvement on build reproducibility. Good stuff!

webmat commented on Rakefile:
```diff
@@ -1,23 +1,19 @@
+end
 
+task :clean do
+  ["vendor/jar-dependencies", "Gemfile.lock"].each do |p|
```
I may be missing some context, but why delete `Gemfile.lock` when cleaning? And why is the file in `.gitignore`?

The lockfile is important to share a very precise known good set of resolved dependencies between developers working on this...

If the issue is that we need to test on multiple incompatible configurations (e.g. JRuby 1.7 & 9.x) and a single Gemfile doesn't cut it, we can always use a the [appraisal](/thoughtbot/appraisal) gem to drive bundler.

jsvd replied:
we don't have the habit of committing the .lock file in the plugins, so the .lock is listed in the .gitignore.

When a plugin is used with logstash, logstash won't care about any locked dependencies besides the ones in the gemspec. So from this perspective, I think that recomputing the .lock on every test run is more likely to catch dependency-related issues that we would see on logstash, which is a good thing.

sihil commented:
Any estimate on when this might get some attention?
